### PR TITLE
Issue #98:  Adds support for month only; month and year only

### PR DESCRIPTION
**Pull request by philgooch**
See #98

Now parses the following constructs:
-   `I am leaving in April`
-  `By June 2004`
-  `Dated: 08/2012`

### CODE REVIEW

File: src/parsers/EN/ENSlashMonthYearFormatParser.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    '([0-9]|0[1-9]|1[012])/([0-9]{4})' + 
+    '(\\s+[^\\d/]|$)', 'i');
+
+var OPENNING_GROUP = 1;
+var ENDING_GROUP = 4;
+
+var MONTH_GROUP = 2;
+var YEAR_GROUP = 3;
+
+exports.Parser = function ENSlashMonthYearFormatParser(argument) {
```
Comment (wanasit) [inline src/parsers/EN/ENSlashMonthYearFormatParser.js]: I think ENSlashMonthFormatParser is a better name.

File: src/parsers/EN/ENMonthNameYearParser.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        ',?\\s*([0-9]{4})(\\s*BE)?' +
+    ')?' +
+    '(?=[^\\s\\w]|$)', 'i');
+
+
+var MONTH_NAME_GROUP = 2;
+var YEAR_GROUP = 3;
+var YEAR_BE_GROUP = 4;
+
+exports.Parser = function ENMonthNameYearParser(){
```
Comment (wanasit) [inline src/parsers/EN/ENMonthNameYearParser.js]: I think just ENMonthNameParser is a better name.

File: src/parsers/EN/ENMonthNameYearParser.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                year = year - 543;
+
+            } else if (year < 100){ 
+
+                year = year + 2000;
+            }
+        }
+
+        if(year){
+            result.start.assign('day', day);
```
Comment (wanasit) [inline src/parsers/EN/ENMonthNameYearParser.js]: Can't we use imply('day', day) instead?

File: src/parsers/EN/ENMonthNameYearParser.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            var nextYear = refMoment.clone().add(1, 'y');
+            var lastYear = refMoment.clone().add(-1, 'y');
+            if( Math.abs(nextYear.diff(moment(ref))) < Math.abs(refMoment.diff(moment(ref))) ){  
+                refMoment = nextYear;
+            }
+            else if( Math.abs(lastYear.diff(moment(ref))) < Math.abs(refMoment.diff(moment(ref))) ){ 
+                refMoment = lastYear;
+            }
+
+            result.start.assign('day', day);
```
Comment (wanasit) [inline src/parsers/EN/ENMonthNameYearParser.js]: same here. should be imply not assign.

File: src/parsers/EN/ENSlashMonthYearFormatParser.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+        var date = null;
+        var year = match[YEAR_GROUP] ;
+        var month = match[MONTH_GROUP];
+        var day   = 1;
+        
+        month = parseInt(month);
+        year = parseInt(year);
+
+        result.start.assign('day', day);
```
Comment (wanasit) [inline src/parsers/EN/ENSlashMonthYearFormatParser.js]: same here

**Comment by philgooch** [comment]
@wanasit thanks for reviewing the PR. I've made the changes suggested, made the regexes a bit more specific so that they handle more cases - additional tests added.

Unfortunately there is now a conflict that requires write access to resolve.

**Comment by wanasit** [comment]
I have merged you PR. Thank you for the improvement.

**Pull request merged**